File: doc-examples/azure-cosmos-example-java/src/main/java/example/Pet.java
package example;

import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public class Pet {

    private String givenName;

    private PetType type;

    public String getGivenName() {
        return givenName;
    }

    public void setGivenName(String givenName) {
        this.givenName = givenName;
    }

    public PetType getType() {
        return type;
    }

    public void setType(PetType type) {
        this.type = type;
    }
}

enum PetType {
    DOG, CAT, HAMSTER
}

File: doc-examples/azure-cosmos-example-python/src/main/python/example/Pet.py
from dataclasses import dataclass
from enum import Enum

from micronaut.serde.annotation import Serdeable


class PetType(Enum):
    DOG = "DOG"
    CAT = "CAT"
    HAMSTER = "HAMSTER"


@Serdeable
@dataclass
class Pet:
    givenName: str | None = None
    type: PetType | None = None
